crates/binstalk/src/fetchers/gh_crate_meta/hosting.rs

```diff
@@ -84,13 +119,17 @@ impl RepositoryHost {
 }
 
 fn apply_filenames_to_paths(
-    paths: &'static [&'static str],
-    filenames: &'static [&'static [&'static str]],
+    paths: &'static [Template<'static>],
+    filenames: &'static [&'static [Template<'static>]],
     suffix: &'static str,
-) -> impl Iterator<Item = String> + Clone + 'static {
+) -> impl Iterator<Item = Template<'static>> + Clone + 'static {
     filenames
         .iter()
         .flat_map(|fs| fs.iter())
         .cartesian_product(paths.iter())
-        .map(move |(filename, path)| format!("{path}/{filename}{suffix}"))
+        .map(move |(filename, path)| {
+            let mut template = path.clone() + filename.clone();
+            template.items.to_mut().push(Item::Text(suffix));
```

Comment:
We could make that nicer by adding an `Add<Template, Item>` impl

Reply:
I think we should also add an `AddAssign` implementation, which will enable the allocated `Template` to be reused.

Reply:
Also I think that should be `path + "/" + filename`?

Reply:
> Also I think that should be `path + "/" + filename`?

The `filename` now contains the slash `/`

Reply:
I've implemented:
 - `impl<'s, 'rhs: 's> ops::AddAssign<&Template<'rhs>> for Template<'s>`
 - `impl<'s, 'rhs: 's> ops::AddAssign<Template<'rhs>> for Template<'s>`
 - `impl<'s, 'item: 's> ops::AddAssign<Item<'item>> for Template<'s>`
 - `impl<'s, 'item: 's> ops::AddAssign<&Item<'item>> for Template<'s>`
 - `impl<'s, 'rhs: 's> ops::Add<Template<'rhs>> for Template<'s>` (improved existing `Add` impl)
 - `impl<'s, 'rhs: 's> ops::Add<&Template<'rhs>> for Template<'s>`
 - `impl<'s, 'item: 's> ops::Add<Item<'item>> for Template<'s>`
 - `impl<'s, 'item: 's> ops::Add<&Item<'item>> for Template<'s>`